### ISSUE TITLE
Add better testing and error messages when calling start plugin methods

### ISSUE THREAD

**Issue by augusto-garcia-afterverse** [mopub/mopub-unity-sdk#189]
One line summary of the issue here

- [x] I am submitting a bug report for existing functionality
- [x] I visited https://developers.mopub.com/ and found no answer
- [x] I checked https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed
- [x] I checked to make sure that this issue has not already been filed

#### MoPub SDK Version:
5.16.4
#### Device model and OS Version:
Any
#### Unity Version:
2019.4.17f
#### Ad Unit IDs used in reproducing the issue:
`<920b6145fb1546cf8b5cf2ac34638bb7>`
#### Steps to reproduce the behavior:
Please list all relevant steps to reproduce the observed behavior.
Mistakenly initialize an ad unit id for multiple types. As an example, calling:

```
	MoPub.LoadRewardedVideoPluginsForAdUnits(new[] {"920b6145fb1546cf8b5cf2ac34638bb7"});
	MoPub.LoadInterstitialPluginsForAdUnits(new[] {"920b6145fb1546cf8b5cf2ac34638bb7"});
	MoPub.LoadBannerPluginsForAdUnits(new[] {"920b6145fb1546cf8b5cf2ac34638bb7"});
```

Will generate an unexpected behaviour on Android, calling for example "hasRewardedVideo" for this ad unit, generating a stacktrace like this:

```
E/Unity: AndroidJavaException: java.lang.NoSuchMethodError: no non-static method with name='hasRewardedVideo' signature='()Z' in class Ljava.lang.Object;
    java.lang.NoSuchMethodError: no non-static method with name='hasRewardedVideo' signature='()Z' in class Ljava.lang.Object;
        at com.unity3d.player.ReflectionHelper.getMethodID(Unknown Source:163)
        at com.unity3d.player.UnityPlayer.nativeRender(Native Method)
        at com.unity3d.player.UnityPlayer.access$300(Unknown Source:0)
        at com.unity3d.player.UnityPlayer$e$1.handleMessage(Unknown Source:95)
        at android.os.Handler.dispatchMessage(Handler.java:101)
        at android.os.Looper.loop(Looper.java:164)
        at com.unity3d.player.UnityPlayer$e.run(Unknown Source:20)
      at UnityEngine.AndroidJNISafe.CheckException () [0x00000] in <00000000000000000000000000000000>:0 
      at UnityEngine.AndroidJNISafe.CallStaticObjectMethod (System.IntPtr clazz, System.IntPtr methodID, 
```

This happens because there is only one dictionary for all ad units and no protection for adding the same id again for a different ad unit type.

#### Expected behavior:

Some kind of error message or exception when attempting to initialize repeated ad units.

#### Observed behavior:

There is no error message or exception being thrown in this case.

#### Evidence:

Already included in the explanation.

**Comment by dravyan** [comment]
@augusto-garcia-playkids, thank you for the feedback. We will relay this to the internal team and see how we can improve in future releases.

Thanks,
MoPub Team

**Comment by chauduyphanvu** [comment]
@augusto-garcia-afterverse, this issue has been addressed in [v5.18.0](https://github.com/mopub/mopub-unity-sdk/releases/tag/v5.18.0) of the MoPub Unity SDK. Please update and let us know if you continue seeing the issue. Thanks!
